Adding support for Basler cameras

emielsteerneman opened the pull request:
Adding support for Basler cameras

joydeep-b commented on src/app/stacks/multistack_robocup_ssl.cpp:
```diff
@@ -124,14 +124,14 @@ MultiStackRoboCupSSL::~MultiStackRoboCupSSL() {
 void MultiStackRoboCupSSL::UpdateServerSettings(const int port,
                           const string& address,
-                          const string& interface,
+                          const string& iface,
```
Please revert this: it's not relevant to Basler camera support.

joydeep-b commented on src/app/stacks/multistack_robocup_ssl.cpp:
```diff
@@ -124,14 +124,14 @@ MultiStackRoboCupSSL::~MultiStackRoboCupSSL() {
   server->_port = port;
   server->_net_address = address;
-  server->_net_interface = interface;
+  server->_net_interface = iface;
```
Ditto: please revert.

joydeep-b commented on src/app/stacks/multistack_robocup_ssl.cpp:
```diff
@@ -144,12 +144,12 @@ void MultiStackRoboCupSSL::UpdateServerSettings(const int port,
   const string address =
       legacy_network_output_settings->multicast_address->getString();
-  const string interface =
+  const string iface =
```
Ditto: please revert.

joydeep-b commented on src/app/stacks/multistack_robocup_ssl.cpp:
```diff
@@ -144,12 +144,12 @@ void MultiStackRoboCupSSL::UpdateServerSettings(const int port,
       legacy_network_output_settings->ds_multicast_port_old->getInt(),
       address,
-      interface,
+      iface,
```
Ditto: please revert.

joydeep-b commented on src/app/stacks/multistack_robocup_ssl.h:
```diff
@@ -63,7 +63,7 @@ class MultiStackRoboCupSSL : public QObject, public MultiVisionStack {
   void UpdateServerSettings(const int port,
                             const string& address,
-                            const string& interface,
+                            const string& iface,
```
Ditto: please revert.

joydeep-b commented on src/shared/net/netraw.h:
```diff
@@ -63,7 +63,7 @@ class UDP {
 
   bool open(int port = 0, bool share_port_for_multicasting=false, bool multicast_include_localhost=false, bool blocking=false);
-  bool addMulticast(const Address &multiaddr,const Address &interface);
+  bool addMulticast(const Address &multiaddr,const Address &iface);
```
Ditto: please revert.

joydeep-b commented on InstallPackagesUbuntu.sh:
```diff
@@ -1,3 +1,3 @@
 #!/bin/bash
 # These are the packages required for install on Ubuntu systems.
-sudo apt-get install g++ libqt4-dev libeigen3-dev protobuf-compiler libprotobuf-dev libdc1394-22 libdc1394-22-dev cmake libv4l-0
+sudo apt-get install g++ libqt4-dev libeigen3-dev protobuf-compiler libprotobuf-dev libdc1394-22 libdc1394-22-dev cmake libv4l-0 libopencv-dev
```
You should add libopencv-dev to the documentation for the requirements for Basler support. Since it it not required for the core build, it should not go here. Please revert this line.

joydeep-b commented on src/app/gui/glwidget.cpp:
```diff
@@ -239,6 +239,8 @@ void GLWidget::mainDraw() {
 void GLWidget::myGLdraw() {
   glPushAttrib ( GL_ALL_ATTRIB_BITS );
   glMatrixMode ( GL_PROJECTION );
+  glPixelStorei(GL_UNPACK_ALIGNMENT, 1);
```
Please revert this: it's not relevant to Basler camera support.

emielsteerneman commented:
I've reverted the variable names, the OPENGL flags, and the dependency. I've updated the README.md for optional Basler Support. This code has been used by the RoboTeam Twente for the last year. Please let me know what kind of proof you want, and if there is anything else that needs to happen.

joydeep-b commented:
Updated diff looks good. 
@g3force is testing on their system, and can merge once confirmed there are no build or run issues when using other cameras.
